core/src/test/scala-3/scalax/collection/EditingTypedSpec.scala

```diff
@@ -0,0 +1,140 @@
+package scalax.collection
+
+import java.time.DayOfWeek._
+import java.time.LocalTime
+import scala.concurrent.duration._
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.Suites
+import org.scalatest.refspec.RefSpec
+
+import scalax.collection.edges.DiEdge
+import scalax.collection.generic._
+
+import scalax.collection.labeled.aviation._
+
+class EditingTypedSpec
```

Comment:
It's not obvious to me why we would need this Spec in different versions.
Anyway, may I suggest going with two PRs?

1. add `-Xsource:3` with all necessary fixes
2. add cross compilation

We also had some overlapping repository activities, unfortunately. For 1., you could have used `coreTestScala3` sources which are in good shape now. Feel free to take over changes from https://github.com/scala-graph/scala-graph/pull/324.

Btw, I'd prefer moving files such that the Git history is kept.